kiota_abstractions/api_error.py

```diff
@@ -1,4 +1,19 @@
+from typing import Optional
+
+
 class APIError(Exception):
+    """The base class for all API errors."""
+
+    def __init__(self, message: str, status_code: Optional[int] = None) -> None:
```

Comment:
I think the `status_code` should be required since every failed API request has a response status code

Reply:
+1

Reply:
Thank. That makes sense. Updated in https://github.com/microsoft/kiota-abstractions-python/pull/35/commits/1f6e9384d9434dbed02cd2f9375d635151086cc6